app/src/main/java/com/hover/stax/presentation/simcard/SimFragment.kt

```diff
@@ -0,0 +1,127 @@
+package com.hover.stax.presentation.simcard
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.fragment.app.Fragment
+import androidx.navigation.NavDirections
+import com.hover.sdk.actions.HoverAction
+import com.hover.stax.MainNavigationDirections
+import com.hover.stax.R
+import com.hover.stax.accounts.AccountsViewModel
+import com.hover.stax.addChannels.ChannelsViewModel
+import com.hover.stax.databinding.FragmentSimBinding
+import com.hover.stax.domain.model.Account
+import com.hover.stax.home.MainActivity
+import com.hover.stax.home.NavHelper
+import com.hover.stax.hover.AbstractHoverCallerActivity
+import com.hover.stax.presentation.home.BalanceTapListener
+import com.hover.stax.presentation.home.BalancesViewModel
+import com.hover.stax.utils.AnalyticsUtil
+import com.hover.stax.utils.UIHelper
+import com.hover.stax.utils.collectLifecycleFlow
+import org.koin.androidx.viewmodel.ext.android.sharedViewModel
+import timber.log.Timber
+
+class SimFragment : Fragment(), BalanceTapListener {
+
+	private var _binding: FragmentSimBinding? = null
+	private val binding get() = _binding!!
+	private val accountViewModel: AccountsViewModel by sharedViewModel()
```

Comment:
SimFragment should only use one viewmodel.

Reply:
Code updated.